airbyte-ci/connectors/connectors_qa/src/connectors_qa/checks/documentation.py

```diff
@@ -101,31 +115,299 @@ def _run(self, connector: Connector) -> CheckResult:
+                if heading_names_value not in template_headings:
+                    heading_names_index += 1  # go to the next actual header as CONNECTOR_SPECIFIC_HEADINGS can be more than one
+                    continue
+                else:
+                    # if actual header is required go to the next template header to validate actual header order
+                    template_headings_index += 1
+                    continue
+            # strict check that actual header equals template header
+            if heading_names_value == template_headings_value:
+                # found expected header, go to the next header in template and actual headers
+                heading_names_index += 1
+                template_headings_index += 1
+                continue
+            # actual header != template header means that template value is not required and can be skipped
+            if template_headings_value in non_required_heading:
+                # found non-required header, go to the next template header to validate actual header
+                template_headings_index += 1
+                continue
+            # any check is True, indexes didn't move to the next step
+            errors.append(reason_titles_not_match(heading_names_value, template_headings_value, template_headings))
+            return errors
+        # indexes didn't move to the last required one, so some headers are missed
+        if template_headings_index != template_headings_len:
+            errors.append(reason_missing_titles(template_headings_index, template_headings, non_required_heading))
+            return errors
+
         return errors
 
-    def check_sections(self, doc_lines: List[str]) -> List[str]:
+    def _run(self, connector: Connector) -> CheckResult:
+        if not connector.documentation_file_path or not connector.documentation_file_path.exists():
+            return self.fail(
+                connector=connector,
+                message="Could not check documentation structure as the documentation file is missing.",
+            )
+
+        doc_lines = [line.lower() for line in connector.documentation_file_path.read_text().splitlines()]
+        if not doc_lines:
+            return self.fail(
+                connector=connector,
+                message="Documentation file is empty",
+            )
+
+        docs_content = connector.documentation_file_path.read_text().rstrip()
+        errors = self.check_docs_structure(docs_content, connector.name_from_metadata)
+
+        if errors:
+            return self.fail(
+                connector=connector,
+                message=f"Documentation headers ordering/naming doesn't follow guidelines:\n {'. '.join(errors)}",
+            )
+        return self.pass_(
+            connector=connector,
+            message="Documentation guidelines are followed",
+        )
+
+
+class CheckDocumentationDescriptions(CheckDocumentationContent):
```

Comment:
I'm not convinced that `description` is the right word here. It looks more like "documentation section content", isn't it?